### ISSUE TITLE
Intermittent connection reset when hitting legal search page

### ISSUE THREAD

**Issue by adborden** [18F/fec-eregs#80]
Seeing intermittent 500s when hitting the search API. Finally was able to get [a stack trace from the web-app](https://gist.github.com/adborden/e84ae0252992704c70bd91404217281d), but the root cause is still unknown.

I'm unclear if the urllib3 error is related, it happens a few minutes before the connection refused error.

```
2016-04-28T15:23:08.27-0700 [App/0]      ERR INFO:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): fec-dev-api.18f.gov
...
2016-04-28T15:32:12.93-0700 [App/0]      ERR requests.exceptions.ConnectionError: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
2016-04-28T15:32:12.94-0700 [RTR/0]      OUT fec-legal-web.18f.gov - [28/04/2016:22:32:12.896 +0000] "GET /data/legal/search/?search_type=all&search=computer HTTP/1.1" 500 0 13149 "https://fec-legal-proxy.18f.gov/legal-resources/" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/50.0.2661.86 Safari/537.36" 127.0.0.1:30952 x_forwarded_for:"198.144.220.83, 127.0.0.1, 10.10.2.70, 52.22.2.98" x_forwarded_proto:"https" vcap_request_id:ae84c124-341f-4954-4e10-ede65c035fde response_time:0.045380972 app_id:e0404e21-78c7-4e04-b319-170512594836
```

**Comment by ccostino** [comment]
@adborden:  If you see this happen again, check the logs in the `api` as well and see if you notice a `worker reset` message or something like that.  I wonder if the two are related and if one is causing the other.

**Comment by adborden** [comment]
@ccostino there doesn't seem any lines like that in `api`. In fact, there's no evidence of the `/v1/legal/search/` endpoint coming through at all related to the web request.

**Comment by ccostino** [comment]
Ah okay - @noahmanger was running into some 500 errors when testing earlier today and I was noticing them come through on the API side it seemed with worker process resets.

**Comment by adborden** [comment]
Bumping `requests` to 2.10. It comes with an updated bundled `urllib3`. We'll see what happens.

**Comment by adborden** [comment]
Still getting the same error. As a short-term solution, we can [enable retries](http://docs.python-requests.org/en/master/api/?highlight=urllib3#requests.adapters.HTTPAdapter) in `request`.

**Comment by porta-antiporta** [comment]
Retries will be implemented as part of #132.
